next.config.js

```diff
@@ -221,7 +221,7 @@ module.exports = async (phase) => {
     },
     async redirects() {
       return [
-        { source: "/teachers", destination: "/", permanent: true },
+        { source: "/ai", destination: "/", permanent: true },
```

Comment:
Is permanent a good idea here? Would a 307 be better than a 308?

I'm not sure what the browser caching of a 308 might be, [this article](https://makandracards.com/makandra/47304-careful-301-308-redirects) suggests it could be forever.

I'm not sure what will happen, but here is my concern......

A user who has previously visited `/teachers` has the `/teachers` -> `/` redirect permanently cached in their browser. After this change has been made they:

1. Visit `/` and see the AI homepage
2. They actually want teachers so they hit the teachers icon and go to `/teachers`
3. The browser thinks `/teachers` is permanently redirected to `/` and so actions this and again shows them `/` which is the AI homepage

It might be that as all this is handled in the client so the browser cache gets overridden. I'm no expert in this but just wanted to check.

Reply:
As things stand, the `/teachers` -> `/` permanent redirect is in place and so we should expect a limited number of visitors to have hit that redirect already.

Running `curl -I https://www.thenational.academy/teachers` returns the following:

```
HTTP/2 308 
date: Wed, 04 Sep 2024 15:44:31 GMT
content-type: text/plain;charset=UTF-8
**cache-control: no-cache**
cache-status: "Netlify Durable"; fwd=miss
cache-status: "Netlify Edge"; fwd=miss
location: /
netlify-vary: header=x-nextjs-data|x-next-debug-logging,cookie=__prerender_bypass|__next_preview_data
refresh: 0;url=/
strict-transport-security: max-age=31536000; includeSubDomains; preload
x-nf-request-id: 01J6YVAXAH3V2AFBGAZ0VDR5JG
cf-cache-status: EXPIRED
set-cookie: __cf_bm=x7_K_cyHMf3pnp9L6D9ThH.B_PR6tFK2O..iVeZL1Zg-1725464671-1.0.1.1-YgFn8q1NVjRls1gTpz_uqX_PTvj6_APWZvb58IihOwnxCzOKrzJVzbsL.uOHwJGVx8lzdb7KziXRfaTDWr0WIQ; path=/; expires=Wed, 04-Sep-24 16:14:31 GMT; domain=.thenational.academy; HttpOnly; Secure; SameSite=None
content-security-policy: upgrade-insecure-requests; frame-ancestors 'self';
permissions-policy: camera=(), geolocation=(), microphone=()
referrer-policy: strict-origin-when-cross-origin
x-content-type-options: nosniff
x-frame-options: DENY
server: cloudflare
cf-ray: 8bdf19f4fc7d945a-LHR
alt-svc: h3=":443"; ma=86400
```

[MDN docs on `cache-control`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control#no-cache) suggest that a value of `no-cache` tells browsers to revalidate with the origin before using a cached response.

This suggests that the browser should respect the behaviour we set here.